D1371R3.md

```diff
@@ -394,6 +394,13 @@ a `void` result. The return types of all patterns must match. If a trailing retu
 type is provided, all patterns must result in an expression returning a type that
 is implicitly convertible to the trailing return type.
 
+If `!` prefix is used before compound statement - the statement would not contribute
+to return type deduction for `inspect` expression. Such a statement is not expected to
+yield a value and should stop the execution either by returning from the enclosing function,
+throwing an exception or terminating the program. This allows users to express desired
```

Comment:
We should specify what the behavior is if the end of the statement is reached.

Reply:
I don't recollect what we decided - call std::terminate?